Object Store/func.py

```diff
@@ -1,67 +1,85 @@
 
-        for lines in data.splitlines():
-            logging.getLogger().info("lines " + lines)
-            payload = {}
-            payload.update({"host":obj})
-            payload.update({"time": eventtime})
+    source = "Oracle Cloud"  # Adding a source name.
+    service = "OCI Logs"     # Adding a service name.
 
-            payload.update({"ddsource":source}) 
-            payload.update({"service":service})
+    datafile = request_one_object(namespace, bucket, resource_name)
+    data = str(datafile, 'utf-8')
 
-            payload.update({"event":lines})
+    # Datadog endpoint URL and token to call the REST interface.
+    # These are defined in the func.yaml file.
+    try:
+        dd_host = os.environ['DATADOG_HOST']
+        dd_token = os.environ['DATADOG_TOKEN']
+        dd_tags = os.environ.get('DATADOG_TAGS', '')
+    except KeyError:
+        err_msg = "Could not find environment variables, \
+                   please ensure DATADOG_HOST and DATADOG_TOKEN \
+                   are set as environment variables."
+        logging.getLogger().error(err_msg)
 
-            
- 
-        headers = {'Content-type': 'application/json', 'DD-API-KEY': datadogtoken}
-        x = requests.post(datadoghost, data = json.dumps(payload), headers=headers)
-        logging.getLogger().info(x.text)
-        print(x.text)
+    for lines in data.splitlines():
+        logging.getLogger().info("lines %s", lines)
+        payload = {}
+        payload.update({"ddsource": source})
+        payload.update({"ddtags": dd_tags})
+        payload.update({"host": resource_name})
+        payload.update({"time": event_time})
+        payload.update({"service": service})
+        payload.update({"event": lines})
 
+    try:
+        headers = {'Content-type': 'application/json', 'DD-API-KEY': dd_token}
+        req = requests.post(dd_host, data=json.dumps(payload), headers=headers)
     except (Exception, ValueError) as ex:
-#        print(str(ex))
         logging.getLogger().info(str(ex))
         return
 
+    logging.getLogger().info(req.text)
+
 
-def request_one_object(namespace, bucket, obj):
-    assert bucket and obj
-    signer = oci.auth.signers.get_resource_principals_signer()
-    object_storage_client = oci.object_storage.ObjectStorageClient(config={}, signer=signer)
-    namespace = namespace
-    bucket_name = bucket
-    object_name = obj
-    get_obj = object_storage_client.get_object(namespace, bucket_name, object_name)
+def request_one_object(namespace: str, bucket: str, resource_name: str) -> bytes:
+    """
+    Calls OCI to request object from Object Storage Client and decompress
+    """
+    oci_signer = oci.auth.signers.get_resource_principals_signer()
+    os_client = oci.object_storage.ObjectStorageClient(config={},
+                                                       signer=oci_signer)
+    get_obj = os_client.get_object(namespace, bucket, resource_name)
     bytes_read = gzip.decompress(get_obj.data.content)
```

Comment:
I may not have all the context, curious: Do we handle any error/excception here caused by the client call?

Reply:
Not at the moment, this was meant to be a cleanup of the current execution without adding too much additional logic to change behavior. There's probably some error catching worth doing here, but will extend that for future work.